crates/db/src/queries/contracts.rs

```diff
@@ -79,22 +79,57 @@ impl DbInner {
         address: Address,
         abi: Option<String>,
         name: Option<String>,
+        proxy_for: Option<Address>,
     ) -> Result<()> {
         let address = format!("0x{:x}", address);
 
-        sqlx::query!(
-            r#" INSERT INTO contracts (address, chain_id, abi, name)
+        if proxy_for.is_some() {
+            let proxy_address = format!("0x{:x}", proxy_for.unwrap());
+
+            sqlx::query!(
+                r#" INSERT INTO contracts (address, chain_id, abi, name, proxy_for)
+                VALUES (?,?,?,?,?)
+                ON CONFLICT(address, chain_id) DO UPDATE SET name=?, abi=?"#,
+                address,
+                chain_id,
+                abi,
+                name,
+                proxy_address,
+                name,
+                abi
+            )
+            .execute(self.pool())
+            .await?;
+
+            sqlx::query!(
+                r#" INSERT INTO contracts (address, chain_id, abi, name, proxied_by)
+                VALUES (?,?,?,?,?)
+                ON CONFLICT(address, chain_id) DO UPDATE SET name=?, abi=?"#,
+                proxy_address,
+                chain_id,
+                abi,
+                name,
+                address,
+                name,
+                abi
```

Comment:
you're adding the same name and abi to both contracts, which is wrong.
the abi/name you're receiving here is suposed to be from the proxy itself, not the implementation